### PR TITLE
VB-613 Refine lead visitor

### PR DESCRIPTION
**Pull request by AledWEvans**
## What does this pull request do?

allows us to have a lead visitor legacy data from nomis with minimal pollution of new system

## What is the intent behind these changes?

Make sure the new system is clean of legacy issues and data structures

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt [hunk 1]
```diff
@@ -21,52 +23,61 @@ import uk.gov.justice.digital.hmpps.visitscheduler.helper.visitNoteCreator
...
-      visitRestriction = VisitRestriction.OPEN,
-      visitContact = CreateContactOnVisitRequest("John Smith", "01234 567890"),
-      visitors = listOf(CreateVisitorOnVisitRequest(123)),
-      visitorSupport = listOf(CreateSupportOnVisitRequest("OTHER", "Some Text")),
-      visitNotes = listOf(
-        VisitNoteDto(type = VISITOR_CONCERN, "My mother in-law is coming"),
-        VisitNoteDto(type = VISIT_OUTCOMES, "My mother wont visit again"),
-        VisitNoteDto(type = VISIT_COMMENT, "Mother in-law should be watched at all times")
+
+    fun createVisitRequest(leadPersonId: Long?): CreateVisitRequest {
```
Comment (philip-milne) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt:52]: I think the original intention was to create a static structure for the test cases, you could achieve the same result by adding 'legacyData = CreateLegacyDataRequest(123)' to the end of the existing definition.

If you want to keep the change you can default the leadPersonId to null and save passing in null values in the call
fun createVisitRequest(leadPersonId: Long? = null): CreateVisitRequest { ....

I'm not a fan of this type of test templating unless its for mocking, might be worth having a backlog item to look into kotlin BDD frameworks.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/data/CreateVisitRequest.kt [hunk 2]
```diff
@@ -34,6 +34,8 @@ data class CreateVisitRequest(
   @Schema(description = "The finishing date and time of the visit", example = "2018-12-01T13:45:00", required = true)
   @NotNull
   val endTimestamp: LocalDateTime,
+  @Schema(description = "Create legacy data", required = false)
```
Comment (philip-milne) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/data/CreateVisitRequest.kt:37]: Great solves the problem

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/jpa/LegacyData.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.Table
+
+@Entity
+@Table(name = "LEGACY_DATA")
+data class LegacyData(
```
Comment (philip-milne) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/jpa/LegacyData.kt:13]: Nicely sandboxed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/jpa/repository/VisitRepository.kt [hunk 4]
```diff
@@ -8,7 +8,7 @@ import uk.gov.justice.digital.hmpps.visitscheduler.jpa.Visit
 import javax.persistence.LockModeType
 
 @Repository
-interface VisitRepository : JpaRepository<Visit, String>, JpaSpecificationExecutor<Visit> {
+interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor<Visit> {
```
Comment (philip-milne) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/jpa/repository/VisitRepository.kt:11]: Good spot

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt [hunk 5]
```diff
@@ -147,6 +154,15 @@ class VisitService(
     )
   }
 
+  private fun saveLegacyData(visit: Visit, leadPersonId: Long) {
```
Comment (philip-milne) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt:157]: Should we rename all of these creates to save?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt [hunk 6]
```diff
@@ -21,52 +23,61 @@ import uk.gov.justice.digital.hmpps.visitscheduler.helper.visitNoteCreator
...
+    }
 
     @Test
     fun `create visit`() {
       webTestClient.post().uri("/visits")
         .headers(setAuthorisation(roles = listOf("ROLE_VISIT_SCHEDULER")))
         .body(
           BodyInserters.fromValue(
-            createVisitRequest
+            createVisitRequest(leadPersonId = null)
```
Comment (philip-milne) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt:80]: Can default if you want

**Review by philip-milne — DISMISSED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt (hunk 5 above)
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt:157]: Yep I tried to make a distinction between the other methods that created objects and did not save!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt (hunk 1 above)
Comment (AledWEvans) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt:52]: Will update

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt (hunk 6 above)
Comment (AledWEvans) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt:80]: will up date

**Review by philip-milne — APPROVED** [review]
LGTM

**Pull request merged**